### PR TITLE
Create fixed-width-integers.md

### PR DESCRIPTION
**Pull request by cppjames**
Created article about fixed width integer types

### CODE REVIEW

File: content/faq/fixed-width-integers.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+disableHLJS: false
+searchHidden: false
+---
+
+Fixed-width integers are integral types with a fixed number of bits. The C++ standard only specifies a minimum byte count for types such as `short`, `int` and `long`. Fixed-width integers guarantee a specific size, but their use can have an impact on portability, since they are not supported by all platforms.
+
+## Fixed-width integer types
+These can be used by including `<cstdint>`. The `intX_t` types are signed integers with exactly `X` bits. `uintX_t` types are the same, but unsigned.
+- `std::int8_t` / `std::uint8_t` (1 byte)
+  - **Note:** These are treated as signed/unsigned characters on most systems. When printing variables of this type, they might show up as their ASCII equivalent.
```
Comment (64) [inline content/faq/fixed-width-integers.md:15]: not entirely sure what you mean by the first sentence here

File: content/faq/fixed-width-integers.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+To optimize integral types for faster access times and to ensure portability with more exotic architectures, the following types allow the compiler to choose a suitable integer size, while also ensuring a minimum number of bits.
+
+For every `std::(u)intX_t` type there is a `std::(u)int_leastX_t` that represents the smallest integral type with at least X bits. These optimize for memory.
+
+Likewise, there are types that optimize for speed: `std::(u)int_fastX_t` (the fastest integral type with at least X bits).
+
+## Fixed-width integer range
+Unsigned integers with X bits range from 0 to 2<sup>X</sup> - 1. Signed integers with X bits typically range from -2<sup>X-1</sup> to 2<sup>X-1</sup> - 1, assuming two's complement representation.
+
+Accurate, platform-specific ranges can also be found in `<cstdint>`:
```
Comment (64) [inline content/faq/fixed-width-integers.md:32]: i think you could mention `numeric_limits` here too

**Comment by 64** [comment]
thanks, just a couple more changes then it’s good to go

**Comment by 64** [comment]
bump

**Review by cppjames — COMMENTED** [review]
Implemented requested changes

File: content/faq/fixed-width-integers.md (hunk 2 above)
Comment (64) [reply content/faq/fixed-width-integers.md:32]: I don't think you changed this

Comment (cppjames) [reply content/faq/fixed-width-integers.md:32]: Hmm, I added some info about `numeric_limits` in the latest commit, maybe I'm missing something

**Comment by 64** [comment]
Thanks again

**Pull request merged**